### PR TITLE
ACMS-1922: Issue with block_content and Acquia CMS Headless.

### PR DESCRIPTION
**Pull request by chandan-singh7929**
**Motivation**
Fixes #[ACMS-1922](https://backlog.acquia.com/browse/ACMS-1922)

**Proposed changes**

**Alternatives considered**

Keep core below 10.1

**Testing steps**

- Setup site with headless starterkit

- Enable block_content module

- Visit /admin/content-models page

- You should see the error.

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ protected function setUp(): void {
    * @dataProvider providerMenu
    */
   public function testChildMenu(string $selector, string $parentMenuName, array $children): void {
-    if ($this->installModule('acquia_cms_toolbar')) {
+    if ($this->installModule('acquia_cms_toolbar') && $this->installModule('block_content')) {
       $this->drupalGet('/admin/headless/dashboard');
       $page = $this->getSession()->getPage();
```
Comment (rajeshreeputra) [inline modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php:85]: @chandan-singh7929 can we check `/admin/content-models` is also accessible in this test and won't throw any error.

**Review by deepakmishra2 — APPROVED** [review]
Changes looks good to me

File: modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php (hunk 1 above)
Comment (chandan-singh7929) [reply modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php:86]: Done

**Review by rajeshreeputra — APPROVED** [review]

**Review by apathak18 — APPROVED** [review]

File: modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php [hunk 2]
```diff
@@ -101,6 +102,12 @@ public function testChildMenu(string $selector, string $parentMenuName, array $c
     }
   }
 
+  public function testContentModelLinks() {
+    // Make sure alias works fine.
+    $this->drupalGet('/admin/content-models');
+    $this->assertSession()->pageTextContains(' Content Models');
```
Comment (deepakmishra2) [inline modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php:108]: @chandan-singh7929 remove extra white before text "Content"

Comment (chandan-singh7929) [reply modules/acquia_cms_headless/tests/src/Functional/PureHeadlessModeMenuTest.php]: Done, thank you!

File: modules/acquia_cms_headless/modules/acquia_cms_headless_ui/src/Service/PureHeadlessModeInstallHandler.php [hunk 3]
```diff
@@ -114,8 +114,17 @@ public function headlessAliases(): array {
...
     ];
+
+    // Due to the changes introduced in version:10.1.0
+    // Block management pages have new paths and menu items.
+    // @see https://www.drupal.org/node/3320855
+    if (version_compare(\Drupal::VERSION, '10.1', '>=')) {
+      unset($aliases['/admin/structure/block/block-content/types']);
+      $aliases['/admin/structure/block-content'] = '/admin/content-models/blocks';
+    }
+    return $aliases;
```
Comment (deepakmishra2) [inline modules/acquia_cms_headless/modules/acquia_cms_headless_ui/src/Service/PureHeadlessModeInstallHandler.php:127]: @chandan-singh7929  add extra line before return

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**